xarray/tests/test_backends.py

```diff
@@ -1040,7 +1043,7 @@ def test_nc4_scipy(self):
             with nc4.Dataset(tmp_file, 'w', format='NETCDF4') as rootgrp:
                 rootgrp.createGroup('foo')
 
-            with self.assertRaisesRegexp(TypeError, 'pip install netcdf4'):
+            with raises_regex(TypeError, 'pip install netcdf4'):
                 open_dataset(tmp_file, engine='scipy')
```

Comment:
@shoyer - any ideas why this test is failing on Appveyor Py3.6: https://ci.appveyor.com/project/shoyer/xray/build/1.0.2844/job/uarr5gsny3wpx9qm#L3326

Reply:
Well, we aren't closing this file properly, so I'm not too surprised that Windows is complaining. I don't know why it's only showing up now, though.

For now, just add `allow_cleanup_failure=True` to the `create_tmp_file()` call. We'll tackle this later in #1668.